gems/smithy-client/lib/smithy-client/plugins/auth.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Client
+    module Plugins
+      # @api private
+      class Auth < Plugin
```

Comment:
I don't think we need another plugin - just make the handler class live inside smithy-client and the code generated auth plugin should register it.